src/main/java/com/manardenza/dao/AbstractDao.java

```diff
@@ -2,27 +2,78 @@
 
 import com.manardenza.entity.AbstractObject;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 import java.util.ArrayList;
 import java.util.List;
 
 public abstract class AbstractDao<T extends AbstractObject> {
 
     protected List<T> database = new ArrayList<>();
+    protected File databaseFile;
+    protected FileOutputStream fos = null;
+    protected ObjectOutputStream oos = null;
+    protected FileInputStream fis = null;
+    protected ObjectInputStream ois = null;
+
+    protected AbstractDao() {
+        databaseFile = new File("src/main/resources/databaseFiles/databaseAbstractFile.bin");
+
+        try {
+            fos = new FileOutputStream(databaseFile);
+            oos = new ObjectOutputStream(fos);
+            fis = new FileInputStream(databaseFile);
+            ois = new ObjectInputStream(fis);
+
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
+        }
+    }
+
+    private void saveToFile() {
+        try {
+            oos.writeObject(database);
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
+        }
+    }
+
+    private void readFromFile() {
+        try {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException ex) { //FIXME: Either log or rethrow this exception.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)